frappe/core/api/file.py

```diff
@@ -27,6 +30,7 @@ def get_attached_images(doctype: str, names: list[str] | str) -> frappe._dict:
 			"attached_to_doctype": doctype,
 			"attached_to_name": ("in", names),
 			"is_folder": 0,
+			"file_type": ("in", file_type),
```

Comment:
FWIW `file_type` guessing as of now isn't guessing popular things like `webp`, so we might have to fix that first before all `registered_extensions` can work.